### PR TITLE
Refactor Concatenate Cubes

### PR DESCRIPTION
**Pull request by meabhnicguidhir**
Attempt to factor out concatenate_cubes and associated private functions from the ensemble_calibration_utilities.

Issue: #51 

Testing:
 - [x] Ran tests and they passed OK
 - Added new tests for the new feature(s) N/A

### CODE REVIEW

File: lib/improver/tests/utilities/test_concatenate_cubes.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Unit tests for the utilities within the `ensemble_calibration_utilities`
```
Comment (gavinevans) [inline lib/improver/tests/utilities/test_concatenate_cubes.py]: I think should say e.g. `within the "cube_manipulation" module`.

File: lib/improver/utilities/cube.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
```
Comment (bayliffe) [inline lib/improver/utilities/cube.py]: Much the same as my thoughts.

File: lib/improver/tests/utilities/test_concatenate_cubes.py (hunk 1 above)
Comment (bayliffe) [inline lib/improver/tests/utilities/test_concatenate_cubes.py]: Second.

Comment (meabhnicguidhir) [inline lib/improver/tests/utilities/test_concatenate_cubes.py]: Done.

File: lib/improver/utilities/cube.py (hunk 2 above)
Comment (meabhnicguidhir) [inline lib/improver/utilities/cube.py]: Changed.

**Pull request merged**